### ISSUE TITLE
Error: Invalid tag "::php::fpm::pool"

### ISSUE THREAD

**Issue by bobweston** [mayflower/puppet-php#129]
I'm trying to get this module working (I'm currently using nodes-php in production, but would like to migrate, if I can get this working)

My hiera looks like this:

<pre>classes:
  - php
php::fpm::pools:
  www:
   listen: '/var/run/php5-fpm.sock' 
   listen_owner: interact
   listen_group: interact
   listen_mode: '0660'
   user: interact
   group: interact
php::settings:
     PHP/upload_max_filesize: '20M'
     PHP/max_file_uploads: '21'	
</pre>	
When I try to spin up and provision the vm (vagrant), I get the following error:

<pre>Error: Invalid tag "::php::fpm::pool" at /tmp/vagrant-puppet/modules-078fcdc8fc841e4b19c52c75025d7110/php/manifests/fpm.pp:51 on node xxxxxxx.org</pre>

That of course points to this

<pre>create_resources(::php::fpm::pool, $real_pools)</pre>

I've tried just invoking it in my manifest by doing include ::php. But that has the same result as setting the class in Hiera.

Any thoughts on what I'm doing wrong? 

I've been banging my head for a couple hours trying to get this working. I am appreciative of any and all help.

I should add that removing all my hiera bits and simply invoking the ::php class <pre>include ::php</pre> results in the same error.

So I'm reasonably certain it is not a problem with the hiera bits.

**Comment by cdenneen** [comment]
Should fix it https://github.com/mayflower/puppet-php/pull/130 hopefully

**Comment by fpletz** [comment]
Hmm, interesting. We never stumbled on that problem.

What version of puppet are you using?

**Comment by cdenneen** [comment]
It's a legacy issue... 

puppet 3.2.3 breaks with:
```puppet
  $real_pools = hiera_hash('php::fpm::pools',  $pools)
  create_resources('::php::fpm::pool', $real_pools)
```
puppet 3.8.3 works fine with it.

Probably for interim I would keep the syntax as:
```puppet
  $real_pools = hiera_hash('php::fpm::pools',  $pools)
  create_resources('php::fpm::pool', $real_pools)
```
which seems to work on all versions of Puppet 3.x

**Comment by globin** [comment]
3.2 is so outdated I'm inclined to say that we don't fix this here and instead encourage you to patch it up for your systems and upgrade as soon as possible..

@fpletz, @fadenb ?

**Comment by cdenneen** [comment]
agreed… it’s actually fixed in 3.3.x so I’ll close PR

**Comment by fpletz** [comment]
Ok, thanks! But we should probably adjust the supported puppet version in `metadata.json`.
